### PR TITLE
Add some benchmarking functionality

### PR DESCRIPTION
**Pull request by dfithian**
I added some benchmarking functionality using [Criterion](https://hackage.haskell.org/package/criterion). It's pretty weird because it's using such an old LTS resolver (current versions of Criterion have features like `envWithCleanup` which would have made this a lot cleaner. Anyway, I ran an echo server using different configured periods and got the results linked in the benchmark.html file. You can see they're all between 50-55 ms. The top graph is the most important; I wouldn't worry too much about the standard dev graphs below since the "outliers" are still below 55 ms.

These were my steps to run the benchmark:
```bash
stack build
./.stack-work/install/x86_64-osx/lts-5.4/7.10.3/bin/benchmark --output benchmark.html --time-limit 5
```
[benchmark.html.zip](https://github.com/creichert/wai-middleware-throttle/files/1162933/benchmark.html.zip)

### CODE REVIEW

File: script/package.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
+name: script
+version: '1.0'
+maintainer: Dan Fithian
+license: AllRightsReserved
```
Comment (creichert) [inline script/package.yaml]: Update the license to match the `LICENSE` file in project root - BSD3

**Review by creichert — APPROVED** [review]
How do you run these with stack? (I could prob figure it out, just checking)

**Comment by dfithian** [comment]
I was unable to run these with stack directly because when I tried to invoke it as a script a la `stack script/benchmark.hs` it failed with a parse error on one of the `#ifndef` directives. The workaround was that I made a package with the `benchmark.hs` file as an executable. The upshot is now that you have to `stack build && ./.stack-work/install/x86_64-osx/lts-5.4/7.10.3/bin/benchmark --output benchmark.html --time-limit 5` (or `stack install && benchmark --output benchmark.html --time-limit 5`. Does that answer your question?

**Comment by creichert** [comment]
@dfithian yea for sure. The old stack resolver is definitely a problem. I started upgrading in #8 but haven't had a chance to finalize it.

Either way, I love benchmarks to this is awesome. Thanks!

**Pull request merged**